### PR TITLE
Fixing lastModified and adding publishReference to Native Model

### PR DESCRIPTION
**Pull request by galiri**
…this will match doc store

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -216,7 +217,6 @@ func handleMessage(msg consumer.Message) {
 }
 
 func constructMessageBody(msg consumer.Message, tid string) (map[string]interface{}, error) {
-
```
Comment (peteclark-ft) [inline app.go]: The TID is re-used for the log messages in the `handleMessage` function, so it seemed cleaner to just re-use the `tid` var given to the `constructMessageBody` func.

File: app.go [hunk 2]
```diff
@@ -225,9 +225,11 @@ func constructMessageBody(msg consumer.Message, tid string) (map[string]interfac
 
 	lastModified, found := msg.Headers["Message-Timestamp"]
 	if !found {
-		warnLogger.Printf("missing time stamp on message", tid)
+		warnLogger.Print("Missing time stamp on message", tid)
```
Comment (khatton-ft) [inline app.go]: In either case is tid actually going to be printed by this log statement?

File: app_test.go [hunk 3]
```diff
@@ -46,8 +48,7 @@ func TestConstructBodyMessageError(t *testing.T) {
 	}
 
 	_, err := constructMessageBody(incomingMsg, "nativeIngesterTest")
-
-	assert.NotNil(t, err)
+	assert.Error(err)
```
Comment (khatton-ft) [inline app_test.go]: This has really thrown me. I don't think it's a good idea to name a variable the same as a package name.

**Comment by khatton-ft** [comment]
I'm not sure it was wise to join this latest change into the same PR, it no longer matches the description.

File: app_test.go (hunk 3 above)
Comment (peteclark-ft) [inline app_test.go]: This is the way Galia prefers to use the assert package - which is also the way the [docs recommend](https://godoc.org/github.com/stretchr/testify/assert).

Comment (khatton-ft) [inline app_test.go]: Fair enough, it's just that I am unfamiliar with that usage and it confused me as I couldn't find that version of the assert methods easily in the docs.

**Pull request merged**